Libraries/Text/RCTTextField.m

```diff
@@ -173,16 +192,19 @@ - (void)textFieldSubmitEditing
 
 - (void)textFieldBeginEditing
 {
-  if (_selectTextOnFocus) {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      [self selectAll:nil];
-    });
-  }
   [_eventDispatcher sendTextEventWithType:RCTTextEventTypeFocus
                                  reactTag:self.reactTag
                                      text:self.text
                                       key:nil
                                eventCount:_nativeEventCount];
+
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
This could potentially race with selection coming from JS right? Should we warn when you use both `selectTextOnFocus` and have controlled selection?